pkg/controllers/scheduling/scheduling_controller.go

```diff
@@ -0,0 +1,185 @@
+	rateLimiter             workqueue.RateLimiter
+	scheduler               scheduler.Scheduler
+}
+
+// NewDecisionSchedulingController return an instance of schedulingController
+func NewSchedulingController(
+	clusterClient clusterclient.Interface,
+	clusterLister clusterlisterv1.ManagedClusterLister,
+	clusterSetLister clusterlisterv1alpha1.ManagedClusterSetLister,
+	clusterSetBindingLister clusterlisterv1alpha1.ManagedClusterSetBindingLister,
+	placementInformer clusterinformerv1alpha1.PlacementInformer,
+	placementDecisionInformer clusterinformerv1alpha1.PlacementDecisionInformer,
+	recorder events.Recorder,
+) factory.Controller {
+	// build controller
+	c := schedulingController{
+		clusterClient:           clusterClient,
+		clusterLister:           clusterLister,
+		clusterSetLister:        clusterSetLister,
+		clusterSetBindingLister: clusterSetBindingLister,
+		placementLister:         placementInformer.Lister(),
+		placementDecisionLister: placementDecisionInformer.Lister(),
+
+		scheduler: core.NewDefaultScheduler(
+			clusterClient,
+			clusterLister,
+			clusterSetLister,
+			clusterSetBindingLister,
+			placementDecisionInformer.Lister(),
+		),
+		rateLimiter: workqueue.NewMaxOfRateLimiter(
+			workqueue.NewItemExponentialFailureRateLimiter(1*time.Second, 120*time.Second),
+			&workqueue.BucketRateLimiter{Limiter: rate.NewLimiter(rate.Limit(10), 100)},
+		),
+	}
+
+	return factory.New().
+		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
+			key, _ := cache.MetaNamespaceKeyFunc(obj)
+			return key
+		}, placementInformer.Informer()).
+		WithFilteredEventsInformersQueueKeyFunc(func(obj runtime.Object) string {
+			accessor, _ := meta.Accessor(obj)
+			labels := accessor.GetLabels()
+			placementName := labels[placementLabel]
+			return fmt.Sprintf("%s/%s", accessor.GetNamespace(), placementName)
+		}, func(obj interface{}) bool {
+			accessor, err := meta.Accessor(obj)
+			if err != nil {
+				return false
+			}
+			labels := accessor.GetLabels()
+			if _, ok := labels[placementLabel]; ok {
+				return true
+			}
+			return false
+		}, placementDecisionInformer.Informer()).
+		// TODO: monitor more resources, like clusters, clustersets and clustersetbindings
+		WithSync(c.sync).
+		ResyncEvery(ResyncInterval).
+		ToController("SchedulingController", recorder)
+}
+
+func (c *schedulingController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	queueKey := syncCtx.QueueKey()
+	namespace, name, err := cache.SplitMetaNamespaceKey(queueKey)
+	if err != nil {
+		// ignore placement whose key is not in format: namespace/name
+		utilruntime.HandleError(err)
+		return nil
+	}
+
+	klog.V(4).Infof("Reconciling placement %q", queueKey)
+	placement, err := c.placementLister.Placements(namespace).Get(name)
+	if errors.IsNotFound(err) {
+		// no work if placement is deleted
+		c.rateLimiter.Forget(queueKey)
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// no work if placement is deleting
```

Comment:
Add to this comment and tell me how stale/old placementdecisions are removed.

Reply:
Since each placementdecision has an owner reference, it will be deleted once its owner is deleted.

Reply:
> Since each placementdecision has an owner reference, it will be deleted once its owner is deleted.

Can you add that to this comment in the code please?